src/pool/cli.c

```diff
@@ -261,6 +261,39 @@ dc_pool_choose_svc_rank(const char *label, uuid_t puuid,
 	return rc;
 }
 
+struct subtract_rsvc_rank_arg {
+	struct pool_domain *srra_nodes;
+	int                 srra_nodes_len;
+};
+
+static bool
+subtract_rsvc_rank(d_rank_t rank, void *varg)
+{
+	struct subtract_rsvc_rank_arg *arg = varg;
+	int                            i;
+
+	for (i = 0; i < arg->srra_nodes_len; i++)
+		if (arg->srra_nodes[i].do_comp.co_rank == rank &&
```

Comment:
Inside this loop, once the rank is found in srra_nodes[], I think the function could then test the status against DC_POOL_SVC_MAP states and either return true or false, and stop the search loop. Right now as coded it looks like the loop will continue even though it might not be necessary.

Reply:
@kccain, thank you for spotting this. Although the code happens to work, I'd consider this a bug.